semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SemanticdbPipeline.scala

```diff
@@ -62,9 +62,38 @@ trait SemanticdbPipeline extends DatabaseOps { self: SemanticdbPlugin =>
         } catch handleError(unit)
       }
 
+      private def synchronizeSourcesAndSemanticdbFiles(): Unit = {
+        val vdb = v.Database.load(Classpath(scalametaTargetroot))
+        val orphanedVentries = vdb.entries.filter(ventry => {
+          val scalaName = v.SemanticdbPaths.toScala(ventry.fragment.name)
+          !config.sourceroot.resolve(scalaName).isFile
+        })
+        orphanedVentries.map(ve => {
+          def cleanupUpwards(file: File): Unit = {
```

Comment:
Unrelated to this PR, but this logic has always bugged me. I think we should try to use idiomatic java.nio wherever possible. Here's how it's possible to delete a directory upwards

```scala
  Files.walkFileTree(
    Paths.get(""),
    new SimpleFileVisitor[Path] {
      override def visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult = {
        if (someLogic() { Files.delete(file) }
        FileVisitResult.CONTINUE
      }
      override def postVisitDirectory(dir: Path, exc: IOException): FileVisitResult = {
        val ds = Files.newDirectoryStream(dir)
        val isEmpty = !ds.iterator().hasNext
        ds.close()
        if (isEmpty) {
          Files.delete(dir)
        }
        FileVisitResult.CONTINUE
      }
    }
  )
```
Sure it's verbose, but by using conventional APIs for IO the code is more approachable for anyone dropping into the codebase.